main/src/main/scala/sbt/internal/server/LanguageServerProtocol.scala

```diff
@@ -102,7 +102,7 @@ private[sbt] object LanguageServerProtocol {
 }
 
 /** Implements Language Server Protocol <https://github.com/Microsoft/language-server-protocol>. */
-private[sbt] trait LanguageServerProtocol extends CommandChannel { self =>
+trait LanguageServerProtocol extends CommandChannel { self =>
```

Comment:
I think we should avoid traits with implementations in API.

Reply:
The important part is just to have access to jsonRpcRespond\*/jsonRpcNotify somehow, this could be achieved by having a trait with abstract defs somewhere.

Reply:
Here's an example code - https://github.com/sbt/sbt/blob/v1.2.8/sbt/src/server-test/handshake/build.sbt

Reply:
IIRC, that was not good enough. I can access `callback` here but I don't have access to the sbt state, so I can't do anything interesting, the only solution I found to that problem when I wrote my prototype bsp plugin a year ago was to indirectly call a command, and have the command call `jsonRpcRespond`, but I can't pass `callback` to that command: https://github.com/dotty-staging/dotty/blob/5eb5bbc1e0fad08647a50690f03ca28131637a6b/sbt-dotty/src/dotty/tools/sbtplugin/DottyBSPPlugin.scala#L193-L197

Reply:
Interesting. I'd have to think a bit more on how can a command or a task send correct RPC response.